skyportal/models/analysis.py

```diff
@@ -553,3 +657,75 @@ def delete_assoc_analysis_data_from_disk(mapper, connection, target):
     for analysis in target.obj_analyses:
         log(f' ... deleting analysis data for analysis id={analysis.id}')
         analysis.delete_data()
+
+
+@event.listens_for(Classification, 'after_insert')
+def create_default_analysis(mapper, connection, target):
+    log(f'Checking for default analyses for classification {target.id}')
+
+    @event.listens_for(inspect(target).session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+        try:
+            from skyportal.handlers.api.analysis import post_analysis
+
+            target_data = target.to_dict()
+
+            stmt = sa.select(DefaultAnalysis).where(
+                DefaultAnalysis.source_filter['classifications'].contains(
+                    [{"name": target_data['classification']}]
+                )
+            )
+            default_analyses = session.scalars(stmt).all()
+
+            for default_analysis in default_analyses:
+                classification_filter = next(
+                    (
+                        classification
+                        for classification in default_analysis.source_filter[
+                            "classifications"
+                        ]
+                        if classification["name"] == target_data['classification']
+                    ),
+                    None,
+                )
+                if classification_filter['probability'] <= target_data['probability']:
+                    log(
+                        f'Creating default analysis {default_analysis.analysis_service.name} for classification {target.id}'
+                    )
```

Comment:
This doesn't seem to terminate if the probability is low, as I would have expected.

Reply:
What I had in mind is that the probability of a new classification has to be over the value you defined in the triggered analysis to trigger an analysis. Do you mean that it should be the opposite? That it should be a maximum value, rather than a minimum? That seems counter intuitive to me. 

Or maybe you meant that in general we shouldn't trigger the analysis if the value is too low anyway, independently of the value the user picked for his triggered analysis?

Reply:
My bad: I thought that `with DBSession() as db_session:` was on the same indentation as the `if` statement. I see what you're doing here.